ton_client/src/debot/run_output.rs

```diff
@@ -70,14 +77,38 @@ impl RunOutput {
                 let std_addr = msg.0.dst().unwrap_or_default();
                 let addr = std_addr.to_string();
                 let wc_and_addr: Vec<&str> = addr.split(':').collect();
-                self.calls.push_back(DebotCallType::Interface {
-                    msg: msg.1.clone(),
-                    id: wc_and_addr
-                        .get(1)
-                        .map(|x| x.to_owned())
-                        .unwrap_or("0")
-                        .to_string(),
-                });
+
+                let mut msg = msg.0.clone();
+                msg.set_src(self.std_addr.clone());
+                if let Ok(msg_base64) = serialize_object_to_base64(&msg, "message") {
+                    self.calls.push_back(DebotCallType::Interface {
+                        msg: msg_base64,
+                        id: wc_and_addr
+                            .get(1)
+                            .map(|x| x.to_owned())
+                            .unwrap_or("0")
+                            .to_string(),
+                    });
+                }
+                return None;
+            }
+        }
+        Some(msg)
+    }
+
+    fn filter_invoke_call<'a>(
+        &mut self,
+        msg: (&'a Message, String),
+    ) -> Option<(&'a Message, String)> {
+        if msg.0.is_internal() {
+            let wc_id = msg.0.workchain_id().unwrap_or(BASECHAIN_ID);
+            if wc_id == BASECHAIN_ID || wc_id == MASTERCHAIN_ID {
```

Comment:
I suggest check for workchain IDs resrved to debots instead of enumerating all existing workchains

Reply:
Done